atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -95,6 +98,9 @@ public CassandraService(
         this.myLocationSupplier = new HostLocationSupplier(this::getSnitch, config.overrideHostLocation());
         this.blacklist = blacklist;
         this.poolMetrics = poolMetrics;
+
+        Supplier<Map<String, String>> hostnamesByIpSupplier = new HostnamesByIpSupplier(this::getRandomGoodHost);
+        this.hostnameByIpSupplier = Suppliers.memoizeWithExpiration(hostnamesByIpSupplier::get, 2, TimeUnit.MINUTES);
```

Comment:
is this necessary? is 2min good enough?

Reply:
This is only ever called in the call to refresh pool, which happens (by default) every 2 minutes. Hardcoding to 2 minutes seems reasonable, but I also see an argument to tie it to the `poolRefreshIntervalSeconds` config and make it expire slightly faster to ensure we get a relatively fresh value on each iteration. We definitely do want the memoization since it's called multiple times in a single refresh call.